worker/jobTypes/githubJob.js

```diff
@@ -41,13 +41,97 @@ class GitHubJobClass {
         return false;
     }
 
+    async constructPrefix(isProdDeployJob){        
+      try{
+        //download published branches file to retrieve prefix and check if repo is versioned 
+        const repoObject = {
+          repoOwner: this.currentJob.payload.repoOwner, repoName: this.currentJob.payload.repoName,
+        };
+        const repoContent = await workerUtils.getRepoPublishedBranches(repoObject)
+        const server_user = await this.getUser()
+        let pathPrefix; 
+        if(isProdDeployJob){
+          //versioned repo
+          if(repoContent && repoContent.content.version.active.length > 1){
+            repoContent.content.prefix
+            pathPrefix = `${repoContent.content.prefix}/${this.currentJob.payload.branchName}`; 
+          }
+        }
+        // server staging commit jobs
+        else if(this.currentJob.payload.patch && this.currentJob.payload.patchType === 'commit'){
+          pathPrefix = `${repoContent.content.prefix}/${this.currentJob.user}/${this.currentJob.payload.localBranchName}/${server_user}/master`; 
+        }
+        //mut only expects prefix or prefix/version for versioned repos, have to remove user from staging prefix
+        if(typeof pathPrefix !== 'undefined' && pathPrefix !== null){
+          const mutPrefix = pathPrefix.split(`/${server_user}`)[0];
+          return[pathPrefix, mutPrefix, server_user]
+        }else{
+          return [null, null, server_user]
+        }
+      }catch(error){
+        console.log(error)
+        throw error
+      }
+        
+    }
+
+    async writeEnvFile(isProdDeployJob){
+        try {
+          const [pathPrefix, mutPrefix, server_user] = await this.constructPrefix(isProdDeployJob)
+          let envVars;
+
+          if(pathPrefix !== null){
+            envVars = 
+            `GATSBY_PARSER_USER=${server_user}
+GATSBY_PARSER_BRANCH=${this.currentJob.payload.branchName}
+PATH_PREFIX=${pathPrefix}
+`;
+          }
+          //front end constructs path prefix for regular githubpush jobs and commitless staging jobs
+          else{
+            envVars = 
+            `GATSBY_PARSER_USER=${server_user}
+GATSBY_PARSER_BRANCH=${this.currentJob.payload.branchName}
+`;
+          }
+    
+          fs.writeFile(`repos/${this.getRepoDirName()}/.env.production`, envVars,  { encoding: 'utf8', flag: 'w' }, function(err) {
+              if(err) {
+                console.log(`error writing .env.production file: ${err.stderr}`);
+                throw err;
+              }
+          }); 
+          //pass mutprefix back to caller to save in prefix field of currentJob, which we pass to stage and deploy targets
+          if(mutPrefix !== null){
+            return mutPrefix
+          }
+          return null
+        } catch (error) {
+         console.log(error)
+         throw error 
+        }
+
+  }
+    async getUser(){
```

Comment:
This function should live somewhere a little more non-build-specific -- in the utils.js perhaps?